code/model/DNReferenceList.php

```diff
@@ -141,13 +141,18 @@ protected function getReferences() {
 			}
 
 			if($this->getTags) {
-				$builds[$reference->getCommitHash()] = DNTag::create($reference, $this->project, $this->data);
```

Comment:
Actually, why can't we just use the tag name as a hash key here? You wouldn't then need to loop through everything down below.